py/mpprint.h

```diff
@@ -80,3 +80,59 @@ int mp_vprintf(const mp_print_t *print, const char *fmt, va_list args);
 #endif
 
 #endif // MICROPY_INCLUDED_PY_MPPRINT_H
```

Comment:
This is the end of file include guard, and should remain at the end of the file.

Reply:
I find this sort of non-standard coding pattern to be confusing, there's a fair chance a future programmer looking at it will think so too and "fix" it, breaking your intended behaviour without realising it.

I believe the print functionality really should be written / used in a way that doesn't take this.

Keep in mind that all C files are compiled in isolation, they all get their own copies of defines & macros included separately.

Also please don't mark concerns as resolved unless the reviewer says as such. I see now I'm not the first person to raise this particular issue.

Reply:
> Keep in mind that all C files are compiled in isolation, they all get their own copies of defines & macros included separately.

I tried to move `#endif // MICROPY_INCLUDED_PY_MPPRINT_H` to the end of the file
and set MP_DEBUG_PRINT_LEVEL  BEFORE any "MicroPython's *.h" includes in two _.c files.
The MP_DEBUG_PRINT() worked only in one file. :(
It may be `cmake` particularity, I don't know.
I keep it as is.